When I tried to compile this locally, I encountered this error,

```
  MSVC target(s)                           : not found
  Checking for 'clang' (C compiler)        : aarch64-linux-android-clang
  Checking size of pointer                 : not found
  Could not determine pointer size. Use `--target-arch' to manually set the pointer size (32bit or 64bit).
  (complete log in /data/data/com.termux/files/usr/tmp/pip-install-kp8af6fo/pyinstaller_c422e412c7804d3ab6e5b39abeb640fa/bootloader/build/config.log)
  ERROR: Failed compiling the bootloader. Please compile manually and rerun
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> No available output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /home/builder/.termux-build/awscli/build/build/venv/bin/python /home/builder/.termux-build/awscli/build/build/venv/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py prepare_metadata_for_build_wheel /data/data/com.termux/files/usr/tmp/tmpbut7u5ju
  cwd: /data/data/com.termux/files/usr/tmp/pip-install-kp8af6fo/pyinstaller_c422e412c7804d3ab6e5b39abeb640fa
  Preparing metadata (pyproject.toml) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> pyinstaller

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/builder/.termux-build/awscli/src/backends/build_system/__main__.py", line 127, in <module>
    main()
    ~~~~^^
  File "/home/builder/.termux-build/awscli/src/backends/build_system/__main__.py", line 123, in main
    parsed_args.func(parsed_args)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/home/builder/.termux-build/awscli/src/backends/build_system/__main__.py", line 41, in build
    aws_venv = _bootstap_venv(
        parsed_args.build_dir,
        parsed_args.artifact,
        parsed_args.download_deps,
    )
  File "/home/builder/.termux-build/awscli/src/backends/build_system/__main__.py", line 76, in _bootstap_venv
    aws_venv.bootstrap(artifact_type, download_deps)
    ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/builder/.termux-build/awscli/src/backends/build_system/awscli_venv.py", line 56, in bootstrap
    self._install_requirements(PORTABLE_EXE_REQUIREMENTS_LOCK)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/builder/.termux-build/awscli/src/backends/build_system/awscli_venv.py", line 79, in _install_requirements
    self._pip_install(
    ~~~~~~~~~~~~~~~~~^
        ["-r", requirements_file],
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
        cwd=cwd,
        ^^^^^^^^
    )
    ^
  File "/home/builder/.termux-build/awscli/src/backends/build_system/awscli_venv.py", line 135, in _pip_install
    self._utils.run(args, **run_kwargs)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "/home/builder/.termux-build/awscli/src/backends/build_system/utils.py", line 235, in run
    return subprocess.run(args, **kwargs)
           ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "/data/data/com.termux/files/usr/lib/python3.13/subprocess.py", line 577, in run
    raise CalledProcessError(retcode, process.args,
                             output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['/home/builder/.termux-build/awscli/build/build/venv/bin/python', '-m', 'pip', 'install', '-r', PosixPath('/home/builder/.termux-build/awscli/src/requirements/download-deps/portable-exe-lock.txt')]' returned non-zero exit status 1.
make: *** [Makefile:20: build] Error 1
```

have you seen that before, and do you know what could cause it?

--- a/packages/awscli/build.sh
+++ b/packages/awscli/build.sh
@@ -2,12 +2,12 @@ TERMUX_PKG_HOMEPAGE=https://aws.amazon.com/cli
 TERMUX_PKG_DESCRIPTION="Universal Command Line Interface for Amazon Web Services"
 TERMUX_PKG_LICENSE="Apache-2.0"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="2.34.0"
+TERMUX_PKG_VERSION="2.34.37"
 TERMUX_PKG_SRCURL="https://github.com/aws/aws-cli/archive/refs/tags/$TERMUX_PKG_VERSION.tar.gz"
-TERMUX_PKG_SHA256=c51f72d6272f6748c874ae395c66b06433922f7399e6b9e4f4aa4ed5806d5331
+TERMUX_PKG_SHA256=52734f1227a086ec604a85429784683134af24fce9585c6021c75912657f6f64
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_UPDATE_VERSION_REGEXP="\d+.\d+.\d+"
-TERMUX_PKG_DEPENDS="libandroid-posix-semaphore, libandroid-support, libbz2, libexpat, libffi, liblzma, libsqlite, mandoc, openssl, readline, zlib"
+TERMUX_PKG_DEPENDS="libandroid-posix-semaphore, libandroid-support, mandoc"
 TERMUX_PKG_SUGGESTS="groff"
 TERMUX_PKG_BUILD_DEPENDS="aosp-libs, python-pip, cmake, ldd"
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
@@ -19,12 +19,26 @@ PYTHON=$TERMUX_PREFIX/bin/python
 # shellcheck disable=SC2115
 termux_step_pre_configure() {
 	export LDFLAGS+=" -lm"
-	export PIP_NO_BINARY=awscrt
+	export PIP_NO_BINARY="awscrt,pyinstaller"
 	export AWS_CRT_BUILD_FORCE_STATIC_LIBS=1
 
 	if [[ "$TERMUX_ON_DEVICE_BUILD" == "true" ]]; then
 		return
 	fi
+
+	export PIP_VERBOSE=1
+	export PIP_NO_CACHE_DIR=1
+
+	case "$TERMUX_ARCH" in
+		aarch64) PYI_PLATFORM=Linux-64bit-arm ;;
+		arm)     PYI_PLATFORM=Linux-32bit-arm ;;
+		x86_64)  PYI_PLATFORM=Linux-64bit-intel ;;
+		i686)    PYI_PLATFORM=Linux-32bit-intel ;;
+		*)
+			echo "ERROR: Unknown architecture: $TERMUX_ARCH"
+			return 1 ;;
+	esac
+	export PYI_PLATFORM
 
 	local PYTHON_INCLUDE="$TERMUX_PREFIX/include/python$TERMUX_PYTHON_VERSION"
 
@@ -61,7 +75,8 @@ termux_step_pre_configure() {
 	PIP_NO_BINARY=$PIP_NO_BINARY \
 	AWS_CRT_BUILD_FORCE_STATIC_LIBS=$AWS_CRT_BUILD_FORCE_STATIC_LIBS \
 	PIP_NO_INDEX=1 \
-	PIP_VERBOSE=1 \
+	PIP_VERBOSE=$PIP_VERBOSE \
+	PIP_NO_CACHE_DIR=$PIP_NO_CACHE_DIR \
 	AS=$AS \
 	CC=$CC \
 	CPP=$CPP \
@@ -79,6 +94,8 @@ termux_step_pre_configure() {
 	CPPFLAGS='$CPPFLAGS -I$PYTHON_INCLUDE' \
 	CXXFLAGS='$CXXFLAGS' \
 	LDFLAGS='$LDFLAGS' \
+	PYTHON='$TERMUX_PREFIX/bin/python' \
+	PYI_PLATFORM=$PYI_PLATFORM \
 	"
 }
 
@@ -95,7 +112,44 @@ termux_step_configure() {
 }
 
 termux_step_make() {
+	pushd "$TERMUX_PKG_SRCDIR"
+
+	pushd "$TERMUX_PKG_TMPDIR"
+	python3 -m venv venv
+	source venv/bin/activate
+	python3 -m pip install pip-tools
+	popd
+	# required by pyinstaller==6.20.0 https://github.com/pyinstaller/pyinstaller/pull/9399
+	# https://github.com/aws/aws-cli/blob/v2/requirements/download-deps/bootstrap.txt#L1
+	sed -i 's/^setuptools==.*/setuptools==82.0.1/' requirements/download-deps/bootstrap.txt
+	pip-compile \
+		--allow-unsafe \
+		--generate-hashes \
+		--output-file=requirements/download-deps/bootstrap-lock.txt \
+		--unsafe-package=flit-core \
+		--unsafe-package=pip \
+		--unsafe-package=setuptools \
+		--unsafe-package=wheel \
+		requirements/download-deps/bootstrap.txt
+	# Workaround for https://github.com/aws/aws-cli/issues/10145
+	echo hatchling >>requirements/portable-exe-extras.txt
+	# required for pep 738 https://github.com/pyinstaller/pyinstaller/pull/9398
+	# https://github.com/aws/aws-cli/blob/v2/requirements/portable-exe-extras.txt#L1
+	sed -i 's/^pyinstaller==.*/pyinstaller==6.20.0/' requirements/portable-exe-extras.txt
+	pip-compile \
+		--allow-unsafe \
+		--generate-hashes \
+		--output-file=requirements/download-deps/portable-exe-lock.txt \
+		--unsafe-package=flit-core \
+		--unsafe-package=pip \
+		--unsafe-package=setuptools \
+		--unsafe-package=wheel \
+		pyproject.toml \
+		requirements/portable-exe-extras.txt
+	deactivate
+
 	if [[ "$TERMUX_ON_DEVICE_BUILD" == "true" ]]; then
+		popd
 		make
 		return
 	fi
@@ -103,17 +157,30 @@ termux_step_make() {
 	local WHEELHOUSE="$TERMUX_PKG_BUILDDIR/wheelhouse"
 	mkdir -p "$WHEELHOUSE"
 
+	local wheel_arch
+	case "$TERMUX_ARCH" in
+		aarch64) wheel_arch=arm64_v8a ;;
+		arm)     wheel_arch=armeabi_v7a ;;
+		x86_64)  wheel_arch=x86_64 ;;
+		i686)    wheel_arch=x86 ;;
+		*)
+			echo "ERROR: Unknown architecture: $TERMUX_ARCH"
+			return 1 ;;
+	esac
+
 	python3 -m pip download \
 		--dest "$WHEELHOUSE" \
-		--platform "linux_$TERMUX_ARCH" \
+		--platform "android_${TERMUX_PKG_API_LEVEL}_${wheel_arch}" \
 		--python-version "$TERMUX_PYTHON_VERSION" \
 		--implementation cp \
 		--abi cp"${TERMUX_PYTHON_VERSION//.}" \
 		--abi abi3 \
 		--abi none \
 		--no-deps \
-		-r "$TERMUX_PKG_SRCDIR/requirements/download-deps/bootstrap-lock.txt" \
-		-r "$TERMUX_PKG_SRCDIR/requirements/download-deps/portable-exe-lock.txt"
+		-r requirements/download-deps/bootstrap-lock.txt \
+		-r requirements/download-deps/portable-exe-lock.txt
+
+	popd
 
 	termux-proot-run \
 		-b "$TERMUX_STANDALONE_TOOLCHAIN/sysroot/usr/include/android:$TERMUX__PREFIX__INCLUDE_DIR/android" \

--- a/packages/awscli/remove-ios-support.patch
+++ b/packages/awscli/remove-ios-support.patch
@@ -1,0 +1,26 @@
+diff --git a/exe/pyinstaller/aws.spec b/exe/pyinstaller/aws.spec
+index eb35333ec..9cf962bb9 100644
+--- a/exe/pyinstaller/aws.spec
++++ b/exe/pyinstaller/aws.spec
+@@ -11,7 +11,7 @@ aws_a = Analysis(['../../bin/aws'],
+              hiddenimports=[],
+              hookspath=['.'],
+              runtime_hooks=[],
+-             excludes=['cmd', 'code', 'pdb'],
++             excludes=['cmd', 'code', 'pdb', '_ios_support'],
+              win_no_prefer_redirects=False,
+              win_private_assemblies=False,
+              cipher=block_cipher)
+diff --git a/exe/pyinstaller/aws_completer.spec b/exe/pyinstaller/aws_completer.spec
+index 5d114e39b..67bb52c0f 100644
+--- a/exe/pyinstaller/aws_completer.spec
++++ b/exe/pyinstaller/aws_completer.spec
+@@ -12,7 +12,7 @@ completer_a = Analysis(['../../bin/aws_completer'],
+              hiddenimports=[],
+              hookspath=['.'],
+              runtime_hooks=[],
+-             excludes=['cmd', 'code', 'pdb'],
++             excludes=['cmd', 'code', 'pdb', '_ios_support'],
+              win_no_prefer_redirects=False,
+              win_private_assemblies=False,
+              cipher=block_cipher)

--- a/packages/awscli/remove-no-build-isolation.patch
+++ b/packages/awscli/remove-no-build-isolation.patch
@@ -1,0 +1,13 @@
+diff --git a/backends/build_system/awscli_venv.py b/backends/build_system/awscli_venv.py
+index be995bf81..e50deda12 100644
+--- a/backends/build_system/awscli_venv.py
++++ b/backends/build_system/awscli_venv.py
+@@ -77,7 +77,7 @@ class AwsCliVenv:
+
+     def _install_requirements(self, requirements_file, cwd=None):
+         self._pip_install(
+-            ["--no-build-isolation", "-r", requirements_file],
++            ["-r", requirements_file],
+             cwd=cwd,
+         )
+

